### ISSUE TITLE
What am I doing wrong?

### ISSUE THREAD

**Issue by DarkAlchy** [ace-step/ACE-Step-1.5#498]
I have a 4090 and I cannot get this to work at less than 25-30GB when using a lora.  That is insane.

I did my daily git pull today and something broke as I trained another lora.  400 epochs.  fantastic loss.  no epoch worked.  WTH?!?  I tried my old lora I trained a day before.  It worked.  Now to make this even worse when I use a lora in this I am suffering at about 27-29GB so I have to leave the room and come back in 10-15 minutes.  Even when training it is so aweful the training never goes above 1-2% GPU utilization whereas image lora training, or video for that matter, is 80-100% utilization.

Something is very wrong, and a friend went digging about this, and showed the repo to Gemini.  Gemini said "Yeah its the same problem. Basically because it decodes the audio per step, instead of decoding the whole thing and caching it, it has to wait for each step to finish. Plus it is single threaded and serialized not parallelized".  It went on to say other points, but this needs to be treated as the devs for the image/video trainers do or this is just brute forcing it.

My thing is I can't even use this on 24GB with a lora.  That is the crazy part.

**Comment by ChuxiJ** [comment]
Could you please help me test this https://github.com/ace-step/ACE-Step-1.5/pull/499

**Comment by DarkAlchy** [comment]
I will, gladly, but I am unsure what am I supposed to change?  I see the code but it doesn't tell me which file, or anything I can latch on with.

It made mention of a script I do not even have, so I was lost.

<img width="751" height="153" alt="Image" src="https://github.com/user-attachments/assets/12a02467-6918-4cb1-a3f7-0a72df943c71" />

That is all I have there.

**Comment by ChuxiJ** [comment]
You can ask the AI（chatgpt or gemini etc）how to switch the branch to this PR, and then you'll be able to test using the code from this PR.

**Comment by DarkAlchy** [comment]
Will this help with not only training but inference with a lora as well because 23.5GB is the max before it goes south?

As far as switching I already know to git checkout 

<img width="89" height="34" alt="Image" src="https://github.com/user-attachments/assets/605bed33-e157-410a-b834-e917656b09fd" />

**Comment by DarkAlchy** [comment]
Model: F:\ACE-Step-1.5\checkpoints\acestep-5Hz-lm-4B
Device: NVIDIA GeForce RTX 4090
GPU Memory Utilization: 0.759
Low GPU Memory Mode: False
2026-02-12 20:36:22.390 | INFO     | acestep.core.generation.handler.lora.lifecycle:load_lora:45 - VRAM before LoRA load: 18.21GB
2026-02-12 20:36:23.654 | INFO     | acestep.core.generation.handler.lora.lifecycle:load_lora:59 - Base decoder state_dict backed up to CPU (3004.9MB)
2026-02-12 20:36:23.654 | INFO     | acestep.core.generation.handler.lora.lifecycle:load_lora:70 - Loading LoRA adapter from F:\ACE-Step-1.5\lora_output\checkpoints\epoch_400\adapter
2026-02-12 20:36:24.508 | INFO     | acestep.core.generation.handler.lora.lifecycle:load_lora:78 - VRAM after LoRA load: 18.29GB
2026-02-12 20:36:24.530 | INFO     | acestep.core.generation.handler.lora.lifecycle:load_lora:86 - LoRA adapter loaded successfully from F:\ACE-Step-1.5\lora_output\checkpoints\epoch_400\adapter (adapters=['default'], targets=192)
2026-02-12 20:36:24.713 | INFO     | acestep.core.generation.handler.lora.controls:set_use_lora:32 - LoRA adapter enabled

**Comment by DarkAlchy** [comment]
I think it worked, now if I can get a training to work.

**Comment by ChuxiJ** [comment]
From your logs, it looks like loading a LoRA with the latest code doesn't increase your VRAM too much.

Could you test the training again? It's hard for me to test here. If it works, I can help you merge this PR.

**Comment by DarkAlchy** [comment]
Let's make sure I do this right for the training.  I set my activation word (trigger word).  Do I prepend or let it be the sole caption?  I am used to images and video so I prepend but for this maybe I should replace?  I have captions and lyrics.

<img width="603" height="712" alt="Image" src="https://github.com/user-attachments/assets/bbefb353-9060-4fda-ab94-b531dbd84e1f" />

**Comment by DarkAlchy** [comment]
Is that the right way to train this?

**Comment by ChuxiJ** [comment]
Either is fine. First, make sure the function works correctly, then you can fine-tune the parameters. We will release an official LoRA tutorial later, and you can also ask the community in our Discord group.

**Comment by DarkAlchy** [comment]
Alright, just run and test memory consumption and if the 102$ actually goes to 80% that is normal usage for training a lora on a 4090.

**Comment by DarkAlchy** [comment]
Pre processing needs help.  this spiked to 24.6

<img width="294" height="143" alt="Image" src="https://github.com/user-attachments/assets/7381bb4a-9fef-45f7-97ba-2484efec0a47" />

My whole system is sluggish, and when pre processing finished vram is still 24.5GB.

**Comment by DarkAlchy** [comment]
which means this will be a very bad training unless that can be lowered to a max of 23.5GB

**Comment by ChuxiJ** [comment]
Okay, I found the issue. The preprocessing didn't do offloading, which caused OOM. I'll fix it later.
